bigframes/eval.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import dataclasses
+from typing import Optional
+
+import pandas
+import pandas.core.computation.parsing as pandas_eval_parsing
```

Comment:
This is a private method. We'll need to be very careful with this if we want to go with this implementation.

Please instead pull this function into `third_party` so we can avoid surprises.

Reply:
Done. Was even considering vendoring the entire eval implementation. That way, I could get rid of the series.values.dtype calls and also be more resilient to any pandas changes.

Reply:
> Was even considering vendoring the entire eval implementation.

Let's do that. Even though `eval` is public, it does spook me to pass in anything but pandas objects into it.

Reply:
Have now vendored the pandas `eval` implementation